target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf289f.dts

```diff
@@ -0,0 +1,435 @@
+                led-failsafe = &led_internal;
+                led-running = &led_internal;
+                led-upgrade = &led_internal;
+        };
+
+        chosen {
+                /*
+                 * bootargs forced by u-boot bootipq command:
+                 * 'ubi.mtd=rootfs root=mtd:ubi_rootfs rootfstype=squashfs rootwait'
+                 */
+                bootargs-append = " root=/dev/ubiblock0_1";
+        };
+
+        /* This node is used to restart modem module to avoid anomalous behaviours on initial communication  */
+        gpio-restart {
+                compatible = "gpio-restart";
+                gpios = <&tlmm 8 GPIO_ACTIVE_HIGH>;
+        };
+
+        leds {
+                compatible = "gpio-leds";
+
+                led_internal: led-0 {
+                        function = LED_FUNCTION_STATUS;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 10 GPIO_ACTIVE_LOW>;
+                        default-state = "keep";
+                };
+
+                led-1 {
+                        function = LED_FUNCTION_WLAN;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 61 GPIO_ACTIVE_LOW>;
+                        linux,default-trigger = "phy0tpt";
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        linux,code = <KEY_WPS_BUTTON>;
+                        gpios = <&tlmm 68 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        soc {
+                ess-psgmii@98000 {
+                        status = "okay";
+                };
+
+                tcsr@1949000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1949000 0x100>;
+                        qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+                };
+
+                tcsr@194b000 {
+                        /* select hostmode */
+                        compatible = "qcom,tcsr";
+                        reg = <0x194b000 0x100>;
+                        qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+                        status = "okay";
+                };
+
+                ess_tcsr@1953000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1953000 0x1000>;
+                        qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+                };
+
+                tcsr@1957000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1957000 0x100>;
+                        qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+                };
+
+                usb2@60f8800 {

```

Comment:
Same label usage comments apply to USB as well.

Reply:
fixed

Reply:
let me know if it's ok now